src/Cache/BlockEsiCache.php

```diff
@@ -53,19 +56,31 @@ class BlockEsiCache extends VarnishCache
      */
     protected $recorder;
 
+    /**
+     * @var ControllerResolverInterface
+     */
+    protected $resolver;
+
+    /**
+     * @var ArgumentResolverInterface
+     */
+    private $argumentResolver;
+
     /**
      * @param string                       $token            A token
      * @param array                        $servers          An array of servers
      * @param RouterInterface              $router           A router instance
      * @param string                       $purgeInstruction The purge instruction (purge in Varnish 2, ban in Varnish 3)
+     * @param ControllerResolverInterface  $resolver         Controller Resolver
+     * @param ArgumentResolverInterface    $argumentResolver Argument Resolver
      * @param BlockRendererInterface       $blockRenderer    A block renderer instance
      * @param BlockContextManagerInterface $contextManager   Block Context manager
      * @param array                        $managers         An array of managers
      * @param Recorder                     $recorder         The cache recorder to build the contextual key
      */
-    public function __construct($token, array $servers, RouterInterface $router, $purgeInstruction, BlockRendererInterface $blockRenderer, BlockContextManagerInterface $contextManager, array $managers = [], Recorder $recorder = null)
+    public function __construct($token, array $servers, RouterInterface $router, $purgeInstruction, ControllerResolverInterface $resolver, ArgumentResolverInterface $argumentResolver, BlockRendererInterface $blockRenderer, BlockContextManagerInterface $contextManager, array $managers = [], Recorder $recorder = null)
```

Comment:
:thinking: Ummmm that's a BC-break I'm afraid

Reply:
I suggest to support only:
"sonata-project/cache": "^2.0",
"sonata-project/cache-bundle": "^3.0"

WDYT

Reply:
still a BC break, would require new major version

Reply:
Yes I think it's a good idea

Reply:
That could be done, but on the `master` branch.